### PR TITLE
feat: added get user notification endpoint

### PR DESCRIPTION
**Pull request by trevorjob**
## Description
This pull request introduces an endpoint for retrieving the current user's notifications.

The changes include
- Endpoint Addition: Added GET ``` /api/v1/notifications/me ``` for retrieving user notifications.
- Authentication: Ensured the endpoint is secured and only accessible to authenticated users.
- Validation: Implemented checks to ensure the user is authenticated before accessing notifications.
- Error Handling: Added appropriate error messages for invalid requests and authentication failures.
- Testing: Added unit and integration tests to ensure functionality and edge cases are covered.
## Motivation and Context

This feature is essential to allow users to securely retrieve their notifications.

## How Has This Been Tested?

- Unit Tests: Added tests to validate the endpoint responsible for retrieving user notifications.
- Integration Tests: Included end-to-end tests to ensure proper functionality and security.
- Tests Include:
    * Retrieving notifications with a valid authentication token.
    - Handling validation errors and authentication failures.
    - Ensuring proper formatting of response data.

​

## Screenshots (if appropriate - Postman, etc):

​

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
      ​

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: tests/v1/test_get_user_nots.py
Comment (joboy-dev) [inline tests/v1/test_get_user_nots.py:1]: Put this file in the notifications folder in tests/v1

**Review by joboy-dev — CHANGES_REQUESTED** [review]
Make changes

**Comment by trevorjob** [comment]
@joboy-dev i've made the changes

File: api/v1/routes/notification.py [hunk 1]
```diff
@@ -29,6 +29,14 @@ def mark_notification_as_read(
     return success_response(status_code=200, message="Notifcation marked as read")
 
 
+@notification.get("/current-user")
+def get_me(
```
Comment (joboy-dev) [inline api/v1/routes/notification.py:33]: change the name to get_current_user_notifications. Something more descriptive

File: api/v1/routes/notification.py [hunk 2]
```diff
@@ -29,6 +29,14 @@ def mark_notification_as_read(
...
+    current_user: User = Depends(user_service.get_current_user),
+    db: Session = Depends(get_db),
+):
+    data = notification_service.get_me(current_user, db)
```
Comment (joboy-dev) [inline api/v1/routes/notification.py:37]: Do the same here

File: api/v1/services/notification.py
Comment (joboy-dev) [inline api/v1/services/notification.py:1]: Nice

**Review by joboy-dev — CHANGES_REQUESTED** [review]
MAke changes. Sorry I didn't notice earlier

**Comment by trevorjob** [comment]
@joboy-dev done

**Review by Goketech — APPROVED** [review]

**Review by joboy-dev — APPROVED** [review]

**Comment by theijhay** [comment]
@Goketech @joboy-dev  his tests failed and you merged this?
